Source/SwiftLintFramework/Rules/ShorthandOperatorRule.swift

```diff
@@ -33,27 +33,37 @@ public struct ShorthandOperatorRule: ConfigurationProviderRule {
                 "↓foo = foo \(operation) 1\n",
                 "↓foo = foo \(operation) aVariable\n",
                 "↓foo = foo \(operation) bar.method()\n",
+                "↓foo.aProperty = foo.aProperty \(operation) 1\n",
+                "↓self.aProperty = self.aProperty \(operation) 1\n"
+            ]
+
+        } + commutativeOperators.flatMap { operation in
+            [
                 "↓foo = 1 \(operation) foo\n",
                 "↓foo = aVariable \(operation) foo\n",
                 "↓foo = bar.method() \(operation) foo\n",
-                "↓foo = bar.method(param: 1, otherParam: 2) \(operation) foo\n",
-                "↓foo.aProperty = foo.aProperty \(operation) 1\n",
-                "↓self.aProperty = self.aProperty \(operation) 1\n"
+                "↓foo = bar.method(param: 1, otherParam: 2) \(operation) foo\n"
             ]
         }
     )
 
-    private static let allOperators = ["+", "-", "/", "*"]
+    private static let allOperators = ["-", "/"] + commutativeOperators
+    private static let commutativeOperators = ["+", "*"]
 
     private static let pattern: String = {
-        let escapedOperators = allOperators.map { "\\\($0)" }.joined()
-        let operators = "[\(escapedOperators)]"
+        let escaped = { (operators: [String]) -> String in
```

Comment:
You use this pattern to escape characters in OperatorFunctionWhitespaceRule and OperatorUsageWhitespaceRule (and possibly others). Might be worth refactoring this into a helper?